### ISSUE TITLE
Runtime socket.error when creating a fetch_pbd couchdb

### ISSUE THREAD

**Issue by ruffsl** [fetchrobotics/fetch_pbd#5]
I'm trying to use the launch file from starting fetch_pbd node, but I'm getting this `socket.error` error. From looking at the traceback, it looks like its having trouble creating a instance of a couchbd:

``` terminal
Traceback (most recent call last):
  File "/home/ruffsl/workspaces/fetchrobotics/src/fetch_pbd/fetch_pbd_interaction/nodes/pbd_interaction_node.py", line 28, in <module>
    interaction = Interaction()
  File "/home/ruffsl/workspaces/fetchrobotics/src/fetch_pbd/fetch_pbd_interaction/src/fetch_pbd_interaction/interaction.py", line 61, in __init__
    self._im_server)
  File "/home/ruffsl/workspaces/fetchrobotics/src/fetch_pbd/fetch_pbd_interaction/src/fetch_pbd_interaction/session.py", line 67, in __init__
    self._db = self._couch.create('fetch_pbd')
  File "/usr/lib/python2.7/dist-packages/couchdb/client.py", line 193, in create
    self.resource.put_json(validate_dbname(name))
  File "/usr/lib/python2.7/dist-packages/couchdb/http.py", line 405, in put_json
    status, headers, data = self.put(*a, **k)
  File "/usr/lib/python2.7/dist-packages/couchdb/http.py", line 384, in put
    return self._request('PUT', path, body=body, headers=headers, **params)
  File "/usr/lib/python2.7/dist-packages/couchdb/http.py", line 419, in _request
    credentials=self.credentials)
  File "/usr/lib/python2.7/dist-packages/couchdb/http.py", line 239, in request
    resp = _try_request_with_retries(iter(self.retry_delays))
  File "/usr/lib/python2.7/dist-packages/couchdb/http.py", line 205, in _try_request_with_retries
    raise e
socket.error: [Errno 111] Connection refused
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >'
  what():  boost: mutex lock failed in pthread_mutex_lock: Invalid argument
[fetch_pbd_interaction-12] process has died [pid 9659, exit code -6, cmd /home/ruffsl/workspaces/fetchrobotics/src/fetch_pbd/fetch_pbd_interaction/nodes/pbd_interaction_node.py __name:=fetch_pbd_interaction __log:=/home/ruffsl/.ros/log/58547894-f22f-11e6-9460-d8cb8a401fcc/fetch_pbd_interaction-12.log].
log file: /home/ruffsl/.ros/log/58547894-f22f-11e6-9460-d8cb8a401fcc/fetch_pbd_interaction-12*.log
```

Full session output I'll link here:
http://pastebin.com/WFyLanWQ

I poked about the web interface, and it's rendering a display and interfaces, but I can't create any actions as obviously the database is up.

**Comment by ruffsl** [comment]
Ok, found my error. Having never used CouchDB before, I had figured that the `python-couchdb` debian package would have also installed `couchdb` as well. Clearly that wasn't the case, as I found I didn't have the couchdb running, let alone installed. So after apt installing the `couchdb` package, that should configure and startup the CouchDB server. 

A simple way to check that its running is to curl the local host on CouchDB's default port:

```
curl http://127.0.0.1:5984/
```

I'd recommend this package be added as a runtime dependency, and perhaps the instructions note about the check to make sure the CouchDB server is up and running prior to ROS launching. Feel free to close.
